### PR TITLE
add preset config for secrets-store ci

### PR DESCRIPTION
**Pull request by aramase**
- Adds preset config for `secrets-store-csi-driver` CI.

cc @ritazh

### CODE REVIEW

**Comment by aramase** [comment]
Have requested for assistance in [#sig-testing](https://kubernetes.slack.com/archives/C09QZ4DQB/p1585854313060400) to add the secret `azure-secrets-store-cred` to prow cluster

**Comment by spiffxp** [comment]
/hold
Can you put this preset in a file that's deeper down in the config directory?

Based on the naming, I'm guessing: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-presets.yaml

eg: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/kops/kops-presets.yaml

**Comment by aramase** [comment]
@spiffxp that make sense! I'll move it.

**Comment by aramase** [comment]
@spiffxp moved it to `config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-presets.yaml`.

/assign @ritazh

File: config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-presets.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    valueFrom:
+      secretKeyRef:
+        name: azure-secrets-store-cred
+        key: tenantid
+  - name: KEYVAULT_NAME
+    valueFrom:
+      secretKeyRef:
+        name: azure-secrets-store-cred
+        key: keyvaultname
+  - name: KEY_VERSION
```
Comment (ritazh) [inline config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-presets.yaml:35]: Assuming this is to retrieve the secret object from keyvault, are we missing the object name here?

Comment (aramase) [reply config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-presets.yaml:35]: The secret and key name are defaulted in here - https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/test/bats/azure.bats#L12-L13. Do you think we should configure it here and read it?

**Comment by spiffxp** [comment]
Please `/hold cancel` once you've confirmed the secret is present; I'll leave the `/lgtm` to @ritazh

**Comment by aramase** [comment]
@BenTheElder just confirmed the secret `azure-secrets-store-cred` has been added to the `test-pods` namespace. 

/hold cancel

**Pull request merged**